### ISSUE TITLE
[data grid] VoiceOver (JAWS) reads wrong column / row when using column pinning feature

### ISSUE THREAD

**Issue by Tucker-Gilligan** [mui/mui-x#10339]
### Steps to reproduce 🕹

Link to live example:

Steps:
1. navigate to https://mui.com/x/react-data-grid/column-pinning
2. turn on JAWS voiceover
3. turn off virtual PC cursor (insert + z)
4. Navigate to unpinned column

### Current behavior 😯

When navigating the data grid when there is a pinned column, the voiceover utility reads the incorrect row/column number for unpinned columns. For example, when there is one pinned column, navigating to the first  data cell in the second column (first unpinned column), voiceover will read column 1 row 12 (should be column 2 row 2).

### Expected behavior 🤔

Voiceover should read correct row/column number when navigating the grid

### Context 🔦

The context of this issue involves working to bring our implementation of the data grid to meet WCAG AA compliance standards, and to pass VPAT accessibility testing. Current implementation requires a lot of hacking

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

### Order ID or Support key 💳 (optional)

48138

**Comment by cherniavskii** [comment]
Hi @Tucker-Gilligan 
I checked the aria attributes, and the columns are enumerated correctly.
Here's the VoiceOver output on MacOS:

<img width="626" src="https://github.com/mui/mui-x/assets/13808724/333156fb-5375-4124-a057-159260191286">

Could you try using the [`ariaV7` experimental flag](https://mui.com/x/react-data-grid/accessibility/#accessibility-changes-in-v7)? It brings some accessibility improvements, maybe it could affect column pinning as well.

**Comment by Tucker-Gilligan** [comment]
Thank you @cherniavskii , I am able to reproduce the issue in the code sandbox you have provided (images attached). This grid with 11 rows is registering as a table with 30 rows due to the pinned columns. Unfortunately, ariaV7 experimental flag did not resolve this issue.

<img width="1690" alt="Screenshot 2023-09-21 at 2 10 22 PM" src="https://github.com/mui/mui-x/assets/72029209/8cb78aa9-8a99-4419-ab20-74d7cb0d3109">

<img width="1565" alt="Screenshot 2023-09-21 at 2 13 40 PM" src="https://github.com/mui/mui-x/assets/72029209/b74d5bc0-be52-409f-8a17-49d4d9933fe9">

**Comment by romgrk** [comment]
Not sure why JAWS shows 30 rows. We clearly have the proper ARIA props, but it's receiving the count of `[role=row]` elements.

One thing to note, we might be able to solve this with the sticky headers change. We can use stickied cells within a row instead of having them in another container. We might also be able to use CSS hover again.

![image](https://github.com/mui/mui-x/assets/1423607/5579d208-5ef1-43b7-9443-366a74465bc9)

**Comment by Tucker-Gilligan** [comment]
The issue is replicable with MacOS, JAWS, and NVDA.

**Comment by cherniavskii** [comment]
@romgrk 

> We clearly have the proper ARIA props, but it's receiving the count of [role=row] elements.

Yes, I played with it a bit, and there's no way to tell the screen reader that multiple HTML elements are part of the same `row` element.

> We can use stickied cells within a row instead of having them in another container

This is an interesting idea, worth exploring for sure!

**Comment by cherniavskii** [comment]
@Tucker-Gilligan 
> The issue is replicable with MacOS voiceover, JAWS, and NVDA.

Yes, I confirm.
This is the limitation of the column pinning feature (at least its current implementation).
This is not exclusive to MUI X DataGrid - see https://github.com/mui/mui-x/issues/5568#issuecomment-1309431365 and https://www.ag-grid.com/javascript-data-grid/accessibility/#navigation-to-pinned-rowscolumns
We will explore a different approach to column pinning as per https://github.com/mui/mui-x/issues/10339#issuecomment-1730369756

**Comment by romgrk** [comment]
> Yes, I played with it a bit, and there's no way to tell the screen reader that multiple HTML elements are part of the same row element.

I'm always a bit fuzzy on the details, would that be an issue with how the browser exposes its a11y tree or with screen readers (or with ARIA)? We could surface the issue to https://github.com/w3c/aria, they're quite responsive & helpful generally.

**Comment by cherniavskii** [comment]
Hi there!
We have reworked the column headers and improved the horizontal scrolling experience in v7, which is [currently in beta](https://github.com/mui/mui-x/releases)! 🎉
You can try out the new Data Grid in the v7 beta docs: https://next.mui.com/x/react-data-grid/

Upgrading from Data Grid v6? Follow our [migration guide](https://next.mui.com/x/migration/migration-data-grid-v6/).
New to the Data Grid? Follow our [installation guide](https://next.mui.com/x/react-data-grid/getting-started/#installation).
Let us know what you think!

**Comment by romgrk** [comment]
Only partially solved but we should complete this shortly, subscribe to #11949 to be notified of the complete fix.
